### ISSUE TITLE
Potential Security Issue w/ changing IPs

### ISSUE THREAD

**Issue by Pnin1204** [Loopring/dexwebapp#94]
Hi,

I've been logged into the Loopring.io wallet on my brave browser, through the brave browser's native wallet.  I left myself logged in for more than a day, and I wasn't notified that I'd been logged out.  However, for this test, I moved locations and connected to a different network.  Then I made an attempt to cancel my order.  It said that I successfully cancelled my order.

Then I tried to place an order, as the cancel button still remained even though the popup had said that it had successfully cancelled the order.  It said it could not cancel the order, which is fine, except that the order was still showing on the website.

I reloaded the page, and was asked to log in again.  I logged in, on this new IP, and the order had indeed cancelled.

I think this means that even though I'd not been online (my laptop was suspended) for more than a day, and I changed locations, I was still able to execute a cancel order instead of having been logged out after some kind of auto time-out function, or IP change.  It could be a security issue, but at the very least, I think that there should be some time-out function for your logins, if possible, especially if the IP has changed.

Hope this helps.  Really a smooth experience otherwise!

**Comment by dong77** [comment]
I'd quote this thread for your proposal. https://security.stackexchange.com/questions/220364/log-user-out-after-change-of-ip-address

With Loopring, there is no backend verification of user credentials; the so call password is used solely on the client-side to derive an EdDSA keypair to signing offchain requests. When you log in, we re-generate the key-pair and compare the public key with the one you put on the Ethereum block through the account registration or password resets. Your private key never leaves the dapp local storage during your active session.

In other words, there is nothing a middle man can do, such as intercepting-then-fake-your requests (remember all requests will have to sign with your client-side EdDSA private key, and our backend use the public-key for verification).

**Comment by kenpole** [comment]
Thank you for the reply.  It's definitely comforting reading about how the pk stays on the local storage.  Appreciate the time behind your response.  I think it could be a good one for an AMA or public forum.

**Comment by xiaowheat** [comment]
Closed the issue
